File: frontend/src/app/core/services/super-admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { College } from '../models/college.model';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class SuperAdminService {
  private readonly apiUrl = `${environment.apiUrl}/super-admin`;

  constructor(private http: HttpClient) {}

  getDashboardStats(): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/dashboard`);
  }

  getAllColleges(): Observable<College[]> {
    return this.http.get<College[]>(`${this.apiUrl}/colleges`);
  }

  getCollegeAdmins(collegeId?: number): Observable<any[]> {
    const params: any = {};
    if (collegeId) params['collegeId'] = String(collegeId);
    return this.http.get<any[]>(`${this.apiUrl}/college-admins`, { params });
  }

  createCollege(college: Partial<College>): Observable<College> {
    return this.http.post<College>(`${this.apiUrl}/colleges`, college);
  }

  updateCollege(id: number, college: Partial<College>): Observable<College> {
    return this.http.put<College>(`${this.apiUrl}/colleges/${id}`, college);
  }

  deleteCollege(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/colleges/${id}`);
  }

  getAllFaculty(collegeId?: number, departmentId?: number): Observable<any[]> {
    let params: any = {};
    if (collegeId)    params['collegeId']    = String(collegeId);
    if (departmentId) params['departmentId'] = String(departmentId);
    return this.http.get<any[]>(`${this.apiUrl}/faculty`, { params });
  }

  getAllStudents(collegeId?: number, departmentId?: number): Observable<any[]> {
    let params: any = {};
    if (collegeId)    params['collegeId']    = String(collegeId);
    if (departmentId) params['departmentId'] = String(departmentId);
    return this.http.get<any[]>(`${this.apiUrl}/students`, { params });
  }

  getDepartmentsByCollege(collegeId: number): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/departments`, { params: { collegeId: String(collegeId) } });
  }

  getSystemReports(): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/reports`);
  }

  generateToken(role: string, collegeId?: number): Observable<any> {
    const params: any = { role };
    if (collegeId) params['collegeId'] = String(collegeId);
    return this.http.post<any>(`${environment.apiUrl}/tokens/generate`, null, { params });
  }

  getAllTokens(): Observable<any[]> {
    return this.http.get<any[]>(`${environment.apiUrl}/tokens`);
  }
}

File: backend/src/main/java/com/grievance/management/service/SuperAdminService.java
package com.grievance.management.service;

import com.grievance.management.entity.*;
import com.grievance.management.enums.GrievanceStatus;
import com.grievance.management.enums.RiskLevel;
import com.grievance.management.exception.ResourceNotFoundException;
import com.grievance.management.repository.*;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
@RequiredArgsConstructor
public class SuperAdminService {

    private final CollegeRepository          collegeRepository;
    private final DepartmentRepository       departmentRepository;
    private final FacultyRepository          facultyRepository;
    private final StudentRepository          studentRepository;
    private final GrievanceRepository        grievanceRepository;
    private final UserRepository             userRepository;
    private final CollegeAdminRepository     collegeAdminRepository;
    private final RiskAssessmentRepository   riskAssessmentRepository;

    // ── Dashboard ─────────────────────────────────────────────────────────────

    public Map<String, Object> getDashboardStats() {
        long pending = grievanceRepository.countByStatus(GrievanceStatus.SUBMITTED)
                     + grievanceRepository.countByStatus(GrievanceStatus.UNDER_REVIEW)
                     + grievanceRepository.countByStatus(GrievanceStatus.ASSIGNED)
                     + grievanceRepository.countByStatus(GrievanceStatus.IN_PROGRESS);

        long resolved = grievanceRepository.countByStatus(GrievanceStatus.RESOLVED)
                      + grievanceRepository.countByStatus(GrievanceStatus.CLOSED);

        long critical = riskAssessmentRepository.countByRiskLevel(RiskLevel.CRITICAL)
                      + riskAssessmentRepository.countByRiskLevel(RiskLevel.HIGH);

        return Map.of(
                "totalColleges",      collegeRepository.count(),
                "totalStudents",      studentRepository.count(),
                "totalFaculty",       facultyRepository.count(),
                "totalCollegeAdmins", collegeAdminRepository.count(),
                "totalGrievances",    grievanceRepository.count(),
                "pendingGrievances",  pending,
                "resolvedGrievances", resolved,
                "criticalGrievances", critical
        );
    }

    // ── Colleges ──────────────────────────────────────────────────────────────

    public List<College> getAllColleges() {
        return collegeRepository.findAll();
    }

    public College createCollege(College college) {
        return collegeRepository.save(college);
    }

    public College updateCollege(Long id, College updated) {
        College college = collegeRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("College not found: " + id));
        college.setName(updated.getName());
        college.setAddress(updated.getAddress());
        college.setCity(updated.getCity());
        college.setState(updated.getState());
        college.setEmail(updated.getEmail());
        college.setPhoneNumber(updated.getPhoneNumber());
        return collegeRepository.save(college);
    }

    public void deleteCollege(Long id) {
        if (!collegeRepository.existsById(id))
            throw new ResourceNotFoundException("College not found: " + id);
        collegeRepository.deleteById(id);
    }

    // ── College Admins ────────────────────────────────────────────────────────

    public List<CollegeAdmin> getAllCollegeAdmins() {
        return collegeAdminRepository.findAll();
    }

    public List<CollegeAdmin> getCollegeAdminsFiltered(Long collegeId) {
        if (collegeId != null)
            return collegeAdminRepository.findByCollegeId(collegeId);
        return collegeAdminRepository.findAll();
    }

    // ── Faculty ───────────────────────────────────────────────────────────────

    public List<Faculty> getAllFaculty() {
        return facultyRepository.findAll();
    }

    public List<Faculty> getFacultyFiltered(Long collegeId, Long departmentId) {
        if (collegeId != null && departmentId != null)
            return facultyRepository.findByDepartmentId(departmentId);
        if (collegeId != null)
            return facultyRepository.findByCollegeId(collegeId);
        return facultyRepository.findAll();
    }

    // ── Students ──────────────────────────────────────────────────────────────

    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    public List<Student> getStudentsFiltered(Long collegeId, Long departmentId) {
        if (collegeId != null && departmentId != null)
            return studentRepository.findByDepartmentId(departmentId);
        if (collegeId != null)
            return studentRepository.findByCollegeId(collegeId);
        return studentRepository.findAll();
    }

    // ── Departments ───────────────────────────────────────────────────────────

    public List<Department> getDepartmentsByCollege(Long collegeId) {
        return departmentRepository.findByCollegeId(collegeId);
    }

    // ── Reports ───────────────────────────────────────────────────────────────

    public Map<String, Object> getSystemReports() {
        long pending = grievanceRepository.countByStatus(GrievanceStatus.SUBMITTED)
                     + grievanceRepository.countByStatus(GrievanceStatus.UNDER_REVIEW)
                     + grievanceRepository.countByStatus(GrievanceStatus.ASSIGNED)
                     + grievanceRepository.countByStatus(GrievanceStatus.IN_PROGRESS);

        long critical = riskAssessmentRepository.countByRiskLevel(RiskLevel.CRITICAL)
                      + riskAssessmentRepository.countByRiskLevel(RiskLevel.HIGH);

        return Map.of(
                "totalUsers",         userRepository.count(),
                "totalColleges",      collegeRepository.count(),
                "totalGrievances",    grievanceRepository.count(),
                "resolvedGrievances", grievanceRepository.countByStatus(GrievanceStatus.RESOLVED),
                "pendingGrievances",  pending,
                "criticalGrievances", critical
        );
    }
}
